docker/DOCKERSWARM.md

```diff
@@ -0,0 +1,185 @@
+      networks:
+        - backend
+      deploy:
+        mode: global
+        restart_policy:
+          condition: on-failure
+        update_config:
+          parallelism: 1
+          delay: 1m30s
+
+    controller:
+      image: martel/mongo-replica-ctrl:latest
+      volumes:
+        - /var/run/docker.sock:/var/run/docker.sock
+      environment:
+        - OVERLAY_NETWORK_NAME=backend
+        - MONGO_SERVICE_NAME=mongo_mongo
+        - REPLICASET_NAME=rs
+        - MONGO_PORT=27017
+            # - DEBUG=1 #uncomment to debug the script
+      entrypoint: python /src/replica_ctrl.py
+      networks:
+        - backend
+      depends_on:
+        - "mongo"
+      deploy:
+        mode: replicated
+        replicas: 1
+        placement:
+          constraints: [node.role==manager]
+        restart_policy:
+          condition: on-failure
+
+  volumes:
+    # External true ensures that the volume is not re-created if already present
+    mongodata:
+      external: true
+
+  networks:
+    backend:
+      external:
+        name: backend
+  ```
+
+1. Deploy the MongoDB ReplicaSet
+
+  ```bash
+  $ docker stack deploy -c docker-compose-mongo.yml mongo
+  ```
+
+1. Check that the deployment worked out correctly:
+
+  ```bash
+  $ docker service logs -f mongo_controller
+  mongo_controller.1.jx664h87o3ft@ms-manager0    | INFO:__main__:Waiting mongo service (and tasks) (mongo_mongo) to start
+  mongo_controller.1.jx664h87o3ft@ms-manager0    | INFO:__main__:Mongo service is up and running
+  mongo_controller.1.jx664h87o3ft@ms-manager0    | INFO:__main__:No previous valid configuration, starting replicaset from scratch
+  mongo_controller.1.jx664h87o3ft@ms-manager0    | INFO:__main__:replSetInitiate: {'ok': 1.0}
+  mongo_controller.1.jx664h87o3ft@ms-manager0    | INFO:__main__:Primary is: 10.0.0.8
+  mongo_controller.1.jx664h87o3ft@ms-manager0    | INFO:__main__:Primary is: 10.0.0.8
+  mongo_controller.1.jx664h87o3ft@ms-manager0    | INFO:__main__:Primary is: 10.0.0.8
+  mongo_controller.1.jx664h87o3ft@ms-manager0    | INFO:__main__:Primary is: 10.0.0.8
+  mongo_controller.1.jx664h87o3ft@ms-manager0    | INFO:__main__:Primary is: 10.0.0.8
+  mongo_controller.1.jx664h87o3ft@ms-manager0    | INFO:__main__:Primary is: 10.0.0.8
+  mongo_controller.1.jx664h87o3ft@ms-manager0    | INFO:__main__:Primary is: 10.0.0.8
+  mongo_controller.1.jx664h87o3ft@ms-manager0    | INFO:__main__:Primary is: 10.0.0.8
+  mongo_controller.1.jx664h87o3ft@ms-manager0    | INFO:__main__:Primary is: 10.0.0.8
+  ```
+
+### Deploy a Orion Context Broker cluster
+
+1. Create a docker-compose-orion.yml file (or reuse the scripts in the
+   [repository](https://github.com/smartsdk/smartsdk-recipes/tree/master/recipes/data-management/context-broker/ha)
+   that offers additional functionalities such as healthchecks and use separate
+   networks for frontend and backend):
+
+  ```yaml
+  version: '3'
+
+  services:
+
+    orion:
+      image: fiware/orion:1.7.0
```

Comment:
This piece of documentation should survive to the pass of the year ;). Thus, I'll recommend using:

```
      image: fiware/orion:latest
```